sympy/core/expr.py

```diff
@@ -2877,6 +2877,57 @@ def _eval_nseries(self, x, n, logx):
                      nseries calls it.""" % self.func)
                      )
 
+    def aseries(self, x, n=6, bound=0, hierar=False):
+        from sympy.series.gruntz import mrv, rewrite, mrv_leadterm
+        from sympy.functions import exp, log
+
+        if x.is_positive is x.is_negative is None:
+            xpos = C.Dummy('x', positive=True, bounded=True)
+            return self.subs(x, xpos).aseries(xpos, n, bound, hir).subs(xpos, x)
+
+        omega, exps = mrv(self, x)
+        if x in omega:
+            s = self.subs(x, exp(x)).aseries(x, n, bound, hir).subs(x, log(x))
```

Comment:
`hir` is undefined here.

Reply:
Sorry for the late response. 
Have updated everything, and also added tests. ( See the comment below).